### PR TITLE
Adding shortMonthDay format

### PR DESCRIPTION
**Pull request by ChrisKraljevicD2L**
I hope I've done this correctly.

Lessons uses [`react-intl`'s](https://formatjs.io/docs/react-intl/api/#formatdate) `{ month: 'short', day: 'numeric' }` date format for displaying pacing dates. We want to keep supporting that format as we're switching over to using `BrightspaceUI/intl` and displaying more dates on content.

If I need to reach out to someone for correct `shortMonthDay` formatting, please let me know.

### CODE REVIEW

File: lib/dateTime.js [hunk 1]
```diff
@@ -537,20 +537,20 @@ export function getDateTimeDescriptor() {
...
 		case 'ja':
-			dateFormats = ['yyyy年M月d日', 'yyyy年M月d日', 'yyyy/MM/dd', 'yyyy年M月', 'M月d日'];
+			dateFormats = ['yyyy年M月d日', 'yyyy年M月d日', 'yyyy/MM/dd', 'yyyy年M月', 'M月d日', 'M月d日'];
 			dayPeriods = ['午前', '午後'];
 			months[0] = months[1] = ['1 月', '2 月', '3 月', '4 月', '5 月', '6 月', '7 月', '8 月', '9 月', '10 月', '11 月', '12 月'];
 			days[0] = days[1] = days[2] = ['日', '月', '火', '水', '木', '金', '土'];
 			break;
 		case 'ko':
-			dateFormats = ['yyyy년 M월 d일 dddd', 'yyyy년 M월 d일', 'yyyy-MM-dd', 'yyyy년 M월', 'M월 d일'];
+			dateFormats = ['yyyy년 M월 d일 dddd', 'yyyy년 M월 d일', 'yyyy-MM-dd', 'yyyy년 M월', 'M월 d일', 'M월 d일'];
```
Comment (dlockhart) [inline lib/dateTime.js:546]: The CLDR says this should be: `MMM d일`

File: lib/dateTime.js [hunk 2]
```diff
@@ -620,6 +620,7 @@ export function getDateTimeDescriptor() {
 			dateFormats[1] = 'MMM d yyyy';
 			dateFormats[2] = 'yyyy-MM-dd';
 			dateFormats[4] = 'MMMM d';
+			dateFormats[5] = 'MMM d';
```
Comment (dlockhart) [inline lib/dateTime.js:623]: The CLDR doesn't have a special entry here for `fr-CA`, but I'm good either way.

Comment (dlockhart) [inline lib/dateTime.js:623]: The CLDR doesn't have a special entry here for `fr-CA`, but I'm good either way.

File: lib/dateTime.js (hunk 1 above)
Comment (ChrisKraljevicD2L) [reply lib/dateTime.js]: Odd, tests still passed after I changed this without changing the expected date format 🤔

**Pull request merged**